packages/jest-util/src/globsToMatcher.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import micromatch = require('micromatch');
+import type {Config} from '@jest/types';
+import replacePathSepForGlob from './replacePathSepForGlob';
+
+const globsMatchers = new Map<
+  string,
+  {
+    isMatch: (str: string) => boolean;
+    negated: boolean;
+  }
+>();
+
+export default function globsToMatcher(
+  globs: Array<Config.Glob>,
+): (path: Config.Path) => boolean {
+  if (globs.length === 0) {
+    return (_: Config.Path): boolean => false;
```

Comment:
Are these type annotations needed? It's not inferred from the outer function

Reply:
I didn't expect them to be needed, but vscode showed me an eslint error when I didn't have them:

> Missing return type on function. eslint@typescript-eslint/explicit-module-boundary-types

Reply:
Hmm, wonder if that's fixed by the freash 3.2 release? Either https://github.com/typescript-eslint/typescript-eslint/pull/2169 or https://github.com/typescript-eslint/typescript-eslint/pull/2176

Reply:
Yeah, seems promising. It looks like Jest is still on v2.30 though, so I think I'd like to say that updating the eslint plugin is out of scope for this PR if that's alright with you.

Reply:
FWIW the breaking changes seem to be pretty okay for this repo: https://github.com/typescript-eslint/typescript-eslint/releases/tag/v3.0.0

Reply:
yeah, that's out of scope for this 😅

Reply:
@SimenB I'm happy to tackle that upgrade, if you want to create an issue and assign it to me :)

Reply:
Thanks @G-Rath! #10177